addons/knowledge/models/knowledge_article.py

```diff
@@ -111,17 +113,19 @@ def _check_members(self):
                 if len(write_members) == 0:
                     raise ValidationError(_("You must have at least one writer."))
 
+    @api.constrains('partner_id', 'permission')
+    def _check_external_member_permission(self):
+        for member in self:
+            if not member.partner_id.user_ids and not member.permission == 'read':
```

Comment:
wrong, portal users also cannot have write or none. But portal users have users.
The condition would be 
`if not member.partner_id.partner_share and not member.permission == 'read': `